webpack/components/Table/TableWrapper.js

```diff
@@ -27,19 +28,20 @@ const TableWrapper = ({
   const dispatch = useDispatch();
   const { foremanPerPage = 20 } = useForemanSettings();
```

Comment:
I was still getting the infinite loop. I think the problem is that `useForemanSettings` doesn't return an object with a `foremanPerPage` key. (It's just `perPage`.) So it was setting it to the default of 20, but then switching between that and the setting from `metadata.per_page`.

(Try inserting a `console.log(data)` inside `updatePagination` to see what I mean)

Changing it to
```suggestion
  const foremanPerPage = useForemanSettings().perPage;
```

fixed it for me.

Reply:
SWEET! On it!